### ISSUE TITLE
[bug] BG2: Lava pit oddity in AR1401

### ISSUE THREAD

**Issue by caillean7** [gemrb/gemrb#215]
**Description**
This is just a curiosity, but still I thought I report it. The Lava pit in the Shadow Temple (AR1401) hurts characters walking onto it with some fire effect. In GemRB this effect only ever affects the first person who walked on it. Even if someone else walks on it afterwards, it still hurts the first one. Tested also on an unmodded installation.

**GemRB versions**
0.84 and 0.85

**To Reproduce**
- Go to AR1401 and proceed to the lava pit
- let one of your party walk across the lava
- then let another character walk onto the lava
- the fire should hurt the first character you had walking across

**Expected behavior**
Grill the person actually standing in the lava

**Comment by lynxlynxlynx** [comment]
Wow, this is a super clean test case, spletttp.bcs only has:
```
IF
  IsOverMe([ANYONE])
THEN
  RESPONSE #50
    ForceSpell(LastTrigger,TRAP_FLAMESTRIKE)
END
```
Very likely to be the same problem causing the light gem above to be an absolute PITA.

**Comment by lynxlynxlynx** [comment]
The trap itself does not have the resetting property/flag, so we disable the trap and never add the Entered trigger again. A one line fix in Highlightable::TriggerTrap, but it makes me wonder how the original handled this.

Can you try modifying the region (trap) called Muck and set a trap detection difficulty higher than 0. Does the original behave the same way then? What about with trap removal difficulty > 0? If nothing still, what about both?

**Comment by caillean7** [comment]
Okay, if I set removal difficulty to > 0 the trap is acting weird. Sometimes a char gets hurt when walking across, sometimes not. And sometimes they can stand in the lava for 2 minutes and nothing happens and when they get off they get burned. So, the trap is pretty messed up, it looks random how it behaves.

With Is trapped set to No it behaves more or less normal, it just takes really long until the burn effects start, way more than the usual few seconds.

**Comment by lynxlynxlynx** [comment]
Thanks, I was really hoping the second part would work, but it appears things are more complicated. :( Did you try several times or was it a one off?

**Comment by caillean7** [comment]
I tried three times, and each time reloaded a save prior to being to AR1401 before.

**Comment by caillean7** [comment]
I was a bit worried because I'm running the game with wine, if that could have any effect on the weird trap behavior, so I tried the 2nd variant with Is trapped = No on EE, too. On the very first try I noticed the same behavior as on the classic engine, but then I reloaded 3 times, went to the Shadow Temple, and the trap behaved normally. Another couple of tests with the classic: the trap worked normally on my PC, but didn't  affect Imoen at all... so, does this make any sense to you? Like at all? 

So - I'd feel better if someone else could try this out, too, before I tell you something wrong here?!
